packages/devtools_app/test/vm_developer/object_inspector/vm_function_display_test.dart

```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:devtools_app/src/config_specific/ide_theme/ide_theme.dart';
+import 'package:devtools_app/src/screens/vm_developer/vm_developer_common_widgets.dart';
+import 'package:devtools_app/src/screens/vm_developer/vm_function_display.dart';
+import 'package:devtools_app/src/screens/vm_developer/vm_service_private_extensions.dart';
+import 'package:devtools_app/src/shared/globals.dart';
+import 'package:devtools_test/devtools_test.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../vm_developer_test_utils.dart';
+
+void main() {
+  late MockFuncObject mockFuncObject;
+
+  const windowSize = Size(4000.0, 4000.0);
+
+  late Func testFunctionCopy;
+
+  setUp(() {
+    setGlobal(IdeTheme, IdeTheme());
+
+    mockFuncObject = MockFuncObject();
+
+    final funcJson = testFunction.toJson();
+    testFunctionCopy = Func.parse(funcJson)!;
+
+    testFunctionCopy.size = 256;
+    testFunctionCopy.isStatic = true;
+    testFunctionCopy.isConst = true;
+
+    when(mockFuncObject.outlineNode).thenReturn(null);
+    when(mockFuncObject.scriptRef).thenReturn(null);
+    when(mockFuncObject.name).thenReturn(testFunctionCopy.name);
+    when(mockFuncObject.ref).thenReturn(testFunctionCopy);
+    when(mockFuncObject.obj).thenReturn(testFunctionCopy);
+    when(mockFuncObject.script).thenReturn(testScript);
+    when(mockFuncObject.pos).thenReturn(testPos);
+
+    when(mockFuncObject.kind).thenReturn(FunctionKind.ImplicitClosureFunction);
+    when(mockFuncObject.deoptimizations).thenReturn(3);
+    when(mockFuncObject.isOptimizable).thenReturn(true);
+    when(mockFuncObject.isInlinable).thenReturn(true);
+    when(mockFuncObject.hasIntrinsic).thenReturn(false);
+    when(mockFuncObject.isRecognized).thenReturn(false);
+    when(mockFuncObject.isNative).thenReturn(null);
+    when(mockFuncObject.vmName).thenReturn('DartVM');
+    when(mockFuncObject.icDataArray).thenReturn(null);
+
+    when(mockFuncObject.fetchingReachableSize)
+        .thenReturn(ValueNotifier<bool>(false));
+    when(mockFuncObject.reachableSize).thenReturn(testRequestableSize);
+    when(mockFuncObject.fetchingRetainedSize)
+        .thenReturn(ValueNotifier<bool>(false));
+    when(mockFuncObject.retainedSize).thenReturn(testRequestableSize);
+    when(mockFuncObject.retainingPath).thenReturn(
```

Comment:
if this mocking code is duplicated you can create helpers like `mockVmFuncObject` or `mockVmClassObject`. Look at `mockConnectedApp` for an example.

Reply:
Done